src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/util/Utils.java

```diff
@@ -17,10 +47,62 @@ public static void configureMetrics() {
         MetricsConfiguration.CONFIG_MAP.put(CircuitBreakerCollector.class, cdefault);
         MetricsConfiguration.CONFIG_MAP.put(ThreadPoolMetricsCollector.class, cdefault);
         MetricsConfiguration.CONFIG_MAP.put(NodeDetailsCollector.class, cdefault);
-        MetricsConfiguration.CONFIG_MAP.put(NodeStatsMetricsCollector.class, cdefault);
+        MetricsConfiguration.CONFIG_MAP.put(NodeStatsAllShardsMetricsCollector.class, cdefault);
+        MetricsConfiguration.CONFIG_MAP.put(NodeStatsFixedShardsMetricsCollector.class, cdefault);
         MetricsConfiguration.CONFIG_MAP.put(MasterServiceEventMetrics.class, new MetricsConfiguration.MetricConfig(1000, 0, 0));
         MetricsConfiguration.CONFIG_MAP.put(MasterServiceMetrics.class, cdefault);
         
     }
 
+    // These methods are utility functions for the Node Stat Metrics Collectors. These methods are used by both the all
+    // shards collector and the few shards collector.
+
+    /**
+     * This function is copied directly from IndicesService.java in elastic search as the original function is not public
+     * we need to collect stats per shard based instead of calling the stat() function to fetch all at once(which increases
+     * cpu usage on data nodes dramatically).
+     * @param indicesService Indices Services which keeps tracks of the indexes on the node
+     * @param indexShard Shard to fetch the metrics for
+     * @param flags  The Metrics Buckets which needs to be fetched.
+     * @return stats given in the flags param for the shard given in the indexShard param.
+     */
+    public static IndexShardStats indexShardStats(final IndicesService indicesService, final IndexShard indexShard,
+                                                  final CommonStatsFlags flags) {
+        if (indexShard.routingEntry() == null) {
+            return null;
+        }
+
+        return new IndexShardStats(
+                indexShard.shardId(),
+                new ShardStats[]{
+                        new ShardStats(
+                                indexShard.routingEntry(),
+                                indexShard.shardPath(),
+                                new CommonStats(indicesService.getIndicesQueryCache(), indexShard, flags),
+                                null,
+                                null,
+                                null)
+                });
+    }
+
+    public static HashMap<String, IndexShard> getShards() {
+        HashMap<String, IndexShard> shards =  new HashMap<>();
```

Comment:
let's change this to HashMap<ShardId, IndexShard>

Reply:
Yes this would take off the need to generate the UniqueShardIdKey altogether, but it would be better if we decide on having the shardRole in the dimensions as well because then these tasks can be taken up together as a separate PR.